File: ios/WidgetModule.m
//
//  WidgetModule.m
//  Solace
//
//  Created by Admin on 2025-06-08.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

// This exposes the Swift module to React Native
@interface RCT_EXTERN_MODULE(WidgetModule, NSObject)

// This exposes the updateWidget function to React Native
// It now takes a single dictionary argument.
RCT_EXTERN_METHOD(updateWidget:(NSDictionary *)data)

@end

File: ios/WidgetModule.swift
import Foundation
import WidgetKit

@objc(WidgetModule)
class WidgetModule: NSObject {

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }

  @objc
  func updateWidget(_ data: [String: String]) {
    guard let quoteText = data["quoteText"] else {
      print("❌ WidgetModule: quoteText not provided.")
      return
    }
    
    if let dataManager = SharedDataManager() {
        dataManager.saveQuote(quoteText)
    } else {
        print("❌ WidgetModule: Failed to initialize SharedDataManager.")
    }
  }
}
